### ISSUE TITLE
Allow for 'groups' of filters on a single babonus

### ISSUE THREAD

**Issue by rapus95** [krbz999/babonus#309]
Background: I have a bonus which applies to thrown weapons if actually thrown (thus, thrown filter is not enough, needs ranged weapon attack) aswell as to finesse weapons in melee.

I wouldn't like duplicating the bonus because if I ever change the bonus I'll have to change it in multiple places, which feels non-ideal

**Comment by krbz999** [comment]
Major major feature, probably will never happen, would be a massive amount of work - but I'll leave this issue open as a reminder of the idea.

**Comment by rapus95** [comment]
Regarding the amount of work, if instead of modeling multiple filters into a single bonus, how about having multiple bonuses share a single bonus &description tab (aka depending on the description & bonus tab of another babonus, linked via the hash you already provide anyway)? Maybe somehow place a sign that it's an inhereted/linked tab.
Then, next question, how to create such a shared object? Add another copy button for a shared babonus.

So the overview would have something like
```
|Bonus A thrown| [Toggle] [Copy] [Copy linked] [Edit] [Delete]
|Bonus B|        [Toggle] [Copy] [Copy linked] [Edit] [Delete]
```
Hitting [Copy linked] for Bonus A
transforms it to look like
```
|Bonus A thrown| [Toggle] [Copy] [Copy linked] [Edit] [Delete]
|-> versatile|   [Toggle] [Copy] [Edit] [Delete]
|Bonus B|        [Toggle] [Copy] [Copy linked] [Edit] [Delete]
```
hitting [Copy] on a linked entry will create another dependent entry that depends on the same "parent". Could also name that linked copy again, but one of them is enough. Anything within | ... | is user defined name except for the "->".
And on actual evaluation you iterate through the list of babonuses as usually, check for the filters as usually, and whenever a filter holds you have a single additional step of indirection to resolve (aka follow the link) to the babonus entry which you then apply as usually.
Could think about whether it's necessary to prevent multiple filters to trigger the same bonus. But could also just tell people to design the filters mutually exclusive.

**Comment by krbz999** [comment]
I'm afraid your example, which was rather hard to follow, doesn't convey to me how it would be less breaking with regards to both the UI and the api, not to even mention the internal mechanics, but I'd be interested in seeing you make a PR for this.

**Comment by krbz999** [comment]
Closing as not planned due to inactivity.
